packages/connection/src/lib.rs

```diff
@@ -49,22 +49,52 @@ pub fn handle<C: CustomQuery>(
             ))
         }
         SetIsm { ism } => {
-            let ism_addr = deps.api.addr_validate(&ism)?;
-
-            ISM.save(deps.storage, &ism_addr)?;
-
-            Ok(event_to_resp(
-                new_event("set_ism").add_attribute("ism", ism),
-            ))
+            match ism {
+                Some(ism) => {
+                    let ism_addr = deps.api.addr_validate(&ism)?;
+
+                    ISM.save(deps.storage, &ism_addr)?;
+
+                    Ok(event_to_resp(
+                        new_event("set_ism").add_attribute("ism", ism),
+                    ))
+                }
+                None => {
+                    let removed_ism = get_ism(deps.storage)?
+                        .map(|addr| addr.to_string())
+                        .unwrap_or_else(|| "none".to_string());
+
+                    ISM.remove(deps.storage);
+
+                    Ok(event_to_resp(
+                        new_event("unset_ism").add_attribute("ism", removed_ism),
+                    ))
```

Comment:
imo we dont really need to know what ISM was removed in the event
```suggestion
                    ISM.remove(deps.storage);

                    Ok(event_to_resp(
                        new_event("unset_ism")
                    ))
```